x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/field_parameters/relations_parameter.tsx

```diff
@@ -0,0 +1,302 @@
+      [item.parent]: item.children.length === 1 ? item.children[0] : item.children,
+    }),
+    {} as RelationsES
+  );
+
+  return {
+    ...field,
+    relations: relationsSerialized,
+  };
+};
+
+/**
+ * Export custom deserializer to be used when we need to deserialize the data coming from ES
+ * @param field The field to be serialized
+ */
+export const relationsDeserializer = (field: Field): Field => {
+  if (field.relations === undefined) {
+    return field;
+  }
+
+  const relations = field.relations as RelationsES;
+  const relationsDeserialized = Object.entries(relations).map(([parent, children]) => ({
+    parent,
+    children: typeof children === 'string' ? [children] : children,
+  }));
+
+  return {
+    ...field,
+    relations: relationsDeserialized,
+  };
+};
+
+const parentConfig: FieldConfig = {
+  label: i18n.translate('xpack.idxMgmt.mappingsEditor.joinType.relations.parentFieldLabel', {
+    defaultMessage: 'Parent',
+  }),
+  validations: [
+    {
+      validator: fieldValidators.emptyField(
+        i18n.translate(
+          'xpack.idxMgmt.mappingsEditor.joinType.validations.parentIsRequiredErrorMessage',
+          {
+            defaultMessage: 'Specify a parent',
+          }
+        )
+      ),
+    },
+  ],
+};
+
+const childConfig: FieldConfig = {
+  label: i18n.translate('xpack.idxMgmt.mappingsEditor.joinType.relations.childFieldLabel', {
+    defaultMessage: 'Child',
+  }),
+  validations: [
+    {
+      validator: fieldValidators.emptyField(
+        i18n.translate(
+          'xpack.idxMgmt.mappingsEditor.joinType.validations.childIsRequiredErrorMessage',
+          {
+            defaultMessage: 'Specify a child',
+          }
+        )
+      ),
+    },
+  ],
+};
+
+export const RelationsParameter = () => {
+  const renderWarning = () => (
+    <EuiCallOut
+      color="warning"
+      iconType="alert"
+      size="s"
+      title={
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.join.multiLevelsParentJoinWarningTitle"
+          defaultMessage="Using multiple levels of relations to replicate a relational model is not recommended. Each level of relation adds an overhead at query time in terms of memory and computation. You should de-normalize your data if you care about performance. {docsLink}"
+          values={{
+            docsLink: (
+              <EuiLink
+                href={documentationService.getJoinMultiLevelsPerformanceLink()}
+                target="_blank"
+              >
+                {i18n.translate(
+                  'xpack.idxMgmt.mappingsEditor.join.multiLevelsPerformanceDocumentationLink',
+                  {
+                    defaultMessage: 'Learn more.',
+                  }
+                )}
+              </EuiLink>
+            ),
+          }}
+        />
+      }
+    />
+  );
+
+  return (
+    <EditFieldFormRow
+      title={i18n.translate('xpack.idxMgmt.mappingsEditor.relationsTitle', {
+        defaultMessage: 'Relations',
```

Comment:
Michael had a good idea, which was to change this name to "Relationships", which is more human-readable and also aligns with the description text:

> Join fields define parent-child relationships among documents of the same index.

Reply:
Good idea, I will change it 👍